src/btree/bt_cursor.c

```diff
@@ -312,13 +306,13 @@ __wt_cursor_valid(WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint64_t recno, bool *vali
                     tmp_key.data = WT_INSERT_KEY(cbt->ins);
                     tmp_key.size = WT_INSERT_KEY_SIZE(cbt->ins);
                     WT_RET(__btcur_bounds_contains_key(
-                      session, &cbt->iface, &tmp_key, WT_RECNO_OOB, &key_out_of_bounds, NULL));
+                      session, &cbt->iface, &tmp_key, NULL, &key_out_of_bounds, NULL));
                 } else
                     WT_RET(__btcur_bounds_contains_key(
-                      session, &cbt->iface, key, WT_RECNO_OOB, &key_out_of_bounds, NULL));
+                      session, &cbt->iface, key, NULL, &key_out_of_bounds, NULL));
             } else
```

Comment:
We technically don't need the if statement at L302 and L303 anymore, thinking of doing like 
```  if (cbt->ins != NULL) {
        /* Get the insert list key. */
        if (key == NULL && btree->type == BTREE_ROW) {
            tmp_key.data = WT_INSERT_KEY(cbt->ins);
            tmp_key.size = WT_INSERT_KEY_SIZE(cbt->ins);
            WT_RET(__btcur_bounds_contains_key(
                session, &cbt->iface, &tmp_key, NULL, &key_out_of_bounds, NULL));
        } else
            WT_RET(__btcur_bounds_contains_key(
                session, &cbt->iface, key, &cbt->recno, &key_out_of_bounds, NULL));

        /* The key value pair we were trying to return weren't within the given bounds. */
        if (key_out_of_bounds)
            return (0);

        WT_RET(__wt_txn_read_upd_list(session, cbt, cbt->ins->upd));
        if (cbt->upd_value->type != WT_UPDATE_INVALID) {
            if (cbt->upd_value->type == WT_UPDATE_TOMBSTONE)
                return (0);
            *valid = true;
            return (0);
        }
    }

Reply:
Oh nice!

Reply:
[8cd13c0](https://github.com/wiredtiger/wiredtiger/pull/8201/commits/8cd13c0feb687b37c8f94f2bb7ca6ba385a84fde)